draft-ietf-quic-transport.md

```diff
@@ -3804,6 +3804,14 @@ number length, connection ID length, and path MTU.  A receiver MUST accept
 packets containing an outdated frame, such as a MAX_DATA frame carrying a
 smaller maximum data than one found in an older packet.
 
+When a sender declares a packet as lost and either retransmits the contained
+frames or marks them for retransmission, the sender SHOULD avoid subsequent
+retransmission of this information if it later receives an acknowledgement for
+that packet. Doing so requires senders to retain information about packets after
+they are declared lost. A sender can discard this information after a period of
+time elapses, such as three times the PTO (Section 6.2 of {{QUIC-RECOVERY}}), or
```

Comment:
3*PTO is really long in this case, and it can consume a fair bit of memory depending upon implementation details.  I'd suggest an ~RTT, since by that point you're expecting the acknowledgement of the retransmitted data.

Reply:
Generally speaking, I tend to think that 3*PTO is fine, and that is because the size of a sentmap entry (i.e., the amount of memory required to remember what was sent in a packet) is only a tiny fraction of the MTU.

If your send buffer is capable of retaining X packets amount of data, then it does not matter if you use *one* tiny fraction, *two* tiny fractions,  or *three*, for sentmap.

All that said, I agree that it might sound a bit frightening, it does not need to be as long as 3*PTO, and therefore my view is that this is a trade-off issue between simplicity (i.e. reusing 3*PTO timer) vs. less frightening.